src/classes/PMT_PaymentWizard_CTRL.cls

```diff
@@ -537,24 +537,38 @@ public with sharing class PMT_PaymentWizard_CTRL {
     * @return Boolean
     */
     private Boolean hasFeatureAccess() {
-        SObjectType payment = npe01__OppPayment__c.getSObjectType();
-        SObjectType opportunity = Opportunity.getSObjectType();
-        return hasAccessTo(payment, paymentFields())
-            && hasAccessTo(opportunity, opportunityFields());
+        List<SObjectType> requiredObjects = new List<SObjectType>{
+                npe01__OppPayment__c.getSObjectType(),
+                Opportunity.getSObjectType()
+        };
+
+        if (hasRequiredObjectLevelAccessTo(requiredObjects)) {
```

Comment:
What do you think about having two methods instead of one `hasRequiredObjectLevelAccessTo`?

I'm thinking two methods for each of the relevant object types. `hasRequiredPermissionsForPayment` and `hasCreateAccessToOpportunity`?

Reply:
I was on the fence with this one, yeah I can split it up.

Reply:
I was also trying not to call two guard methods in the `if` statement.